### ISSUE TITLE
Type error: Private identifiers are only available when targeting ECMAScript 2015 and higher.

### ISSUE THREAD

**Issue by sagardwivedi** [ferdikoomen/openapi-typescript-codegen#1992]
**Describe the bug**
I have ` "target": "ES2022"` then also i am getting `Type error: Private identifiers are only available when targeting ECMAScript 2015 and higher.`.

I am using `NextJS`

![image](https://github.com/ferdikoomen/openapi-typescript-codegen/assets/70888430/35e96aab-041c-47f1-a975-dc3bd4066225)

**Comment by danilofuchs** [comment]
This seems like a Typescript configuration issue on your side. Please check if your tsconfig with target ES2022 includes the generated folder

**Comment by sagardwivedi** [comment]
I had set `"target": "ES2022"`, but still encountered the same error. I removed the client generation and am now using it in the normal way.

**Comment by astromme** [comment]
I'm also experiencing this issue when using `npm run build` in a Next.JS project.

@sagardwivedi can you share more about what you changed to make it work?

**Comment by willtpwise** [comment]
Same here

Using the default next.js configuration from a brand new next.js app

```

{
  "compilerOptions": {
    "target": "es2017",
    "lib": ["dom", "dom.iterable", "esnext"],
    "allowJs": true,
    "skipLibCheck": true,
    "strict": true,
    "noEmit": true,
    "esModuleInterop": true,
    "module": "esnext",
    "moduleResolution": "bundler",
    "resolveJsonModule": true,
    "isolatedModules": true,
    "jsx": "preserve",
    "incremental": true,
    "plugins": [
      {
        "name": "next"
      }
    ],
    "paths": {
      "@/*": ["./src/*"]
    }
  },
  "include": ["next-env.d.ts", "**/*.ts", "**/*.tsx", ".next/types/**/*.ts"],
  "exclude": ["node_modules"]
}

```

```
Type error: Private identifiers are only available when targeting ECMAScript 2015 and higher.

  24 |
  25 | export class CancelablePromise<T> implements Promise<T> {
> 26 |     #isResolved: boolean;
```

```
"next": "14.1.0",
"openapi-typescript-codegen": "^0.27.0",
"typescript": "^5"
```

**Comment by sagardwivedi** [comment]
I attempted to integrate FastAPI with Next.js for practice. After encountering challenges and searching for solutions, I ultimately switched to Supabase. However, the project has been put on hold due to an upcoming exam.

**Comment by sagardwivedi** [comment]
> I'm also experiencing this issue when using `npm run build` in a Next.JS project.
> 
> @sagardwivedi can you share more about what you changed to make it work?

I attempted to integrate FastAPI and Next.js for practice, but faced challenges and couldn't find satisfactory solutions. Consequently, I decided to switch to Supabase. That's why I closed the issue, not because I found a solution.

**Comment by astromme** [comment]
Good to know. I also decided to switch to an alternative: https://openapi-generator.tech/

**Comment by staklau** [comment]
Would love for this to be reopened as it's still an issue if you don't want to change to another library.

**Comment by sagardwivedi** [comment]
> Would love for this to be reopened as it's still an issue if you don't want to change to another library.

No Problem. I will reopen.

**Comment by mrlubos** [comment]
Anyone able to post a CodeSandbox link? If I can reproduce it, I'll fix it

**Comment by valentin-harrang** [comment]
@sagardwivedi I have the same problem. Have you fixed your problem?

**Comment by mrlubos** [comment]
cc @jordanshatford

**Comment by jordanshatford** [comment]
@valentin-harrang are you able  to share a link to the repository? I can look into this. Check out our form that is more actively maintained if you’d like https://github.com/hey-api/openapi-ts

**Comment by valentin-harrang** [comment]
@jordanshatford thank you for the answer. Unfortunately I can't share the repository, which is a private repository on the Gitlab of the company that employs me. Unfortunately I'm not authorised, but I can share some code files with you if you want. 😊

**Comment by jordanshatford** [comment]
@valentin-harrang if possible the next config file and your tsconfig. The issue likely tracks back to the tsconfig values so I just want to replicate it

**Comment by valentin-harrang** [comment]
@jordanshatford Here are some additional elements:

tsconfig.json:

```json
{
  "compilerOptions": {
    "baseUrl": "./src",
    "lib": ["dom", "dom.iterable", "esnext"],
    "allowJs": true,
    "skipLibCheck": true,
    "strict": true,
    "noEmit": true,
    "esModuleInterop": true,
    "module": "esnext",
    "moduleResolution": "bundler",
    "resolveJsonModule": true,
    "isolatedModules": true,
    "jsx": "preserve",
    "incremental": true,
    "plugins": [
      {
        "name": "next"
      }
    ],
    "paths": {
      "@/*": ["./*"],
    }
  },
  "include": ["next-env.d.ts", "**/*.ts", "**/*.tsx", ".next/types/**/*.ts"],
  "exclude": ["node_modules"]
}
```

next.config.mjs:

```js
import createNextIntlPlugin from "next-intl/plugin";
import ms from "ms";

const withNextIntl = createNextIntlPlugin("./src/config/i18n.ts");

/** @type {import('next').NextConfig} */
const nextConfig = {
  images: {
    remotePatterns: [
      {
        protocol: 'https',
        hostname: 'loremflickr.com',
      },
    ],
  },
  headers() {
    return [
      {
        source: "/((?!_next|favicon.ico).*)",
        missing: [
          {
            type: "header",
            key: "Next-Router-Prefetch",
          },
        ],
        headers: [
          {
            key: "Cache-Control",
            value: [
              `s-maxage=` + ms("1d") / 1000,
              `stale-while-revalidate=` + ms("1y") / 1000,
            ].join(", "),
          },
        ],
      },
    ];
  },
  output: "standalone", // For self-hosting in a Docker container.
  transpilePackages: ["lucide-react"],
};

export default withNextIntl(nextConfig);
```

I added this script int `package.json`:

```
"codegen": "openapi --input ./openapi.json --output ./src/types/generated",
```

And I run the script with : `npm run codegen`

Versions:

Next.js: 14.1.3
openapi-typescript-codegen: 0.28.0
typescript: 5.4.2

**Comment by valentin-harrang** [comment]
@jordanshatford I tried to generate my types thanks to https://github.com/hey-api/openapi-ts but I have the same problem in same file (`CancelablePromise.ts`)

**Comment by jordanshatford** [comment]
@valentin-harrang thanks for testing that. I should have some time later today to check it out and figure out what’s causing it.

**Comment by valentin-harrang** [comment]
@jordanshatford thank you so much :)

**Comment by jordanshatford** [comment]
@valentin-harrang Try settings `"target": "ES6",` (or `"target": "ES2015"`) in your tsconfig. I also found that replacing `#isResolved` with `private isResolved` (same for all other fields) fixes the problem, however I do not know if we want to make that change in the package. [see](https://stackoverflow.com/questions/59641564/what-are-the-differences-between-the-private-keyword-and-private-fields-in-types) for info.

I am discussing with the other maintainer to get his input.

**Comment by jordanshatford** [comment]
@valentin-harrang  Update: we are going to fix the issue in our repository. we dont see any issues converting those values to private instead of `#`.

**Comment by jordanshatford** [comment]
@valentin-harrang  this has been fixed in v0.31.0 of @hey-api/openapi-ts

**Comment by valentin-harrang** [comment]
@jordanshatford Thank you very much for your work and your responsiveness. I've just updated the package and it works perfectly ❤️

**Comment by Lamarcke** [comment]
If anyone's still facing this and don't want to migrate to [@hey-api/openapi-ts](https://github.com/hey-api/openapi-ts), i made a fork of this repo with the fix from https://github.com/hey-api/openapi-ts/pull/171.

Just update your package.json:
from: `"openapi-typescript-codegen": "^0.29.0"`
to: `"@lamarcke/openapi-typescript-codegen": "^0.29.1"`

Source code: [@lamarcke/openapi-typescript-codegen](https://github.com/Lamarcke/openapi-typescript-codegen)

I can't promise i will keep updating this fork, but i will fix issues as they appear for me, since i like the current API and it's being used in most of my projects.

**Comment by mrlubos** [comment]
@Lamarcke is the main reason for not wanting to migrate the difference in generated SDKs?

**Comment by Lamarcke** [comment]
> @Lamarcke is the main reason for not wanting to migrate the difference in generated SDKs?

Yeah basically, it would require a big refactor in my side, since the newer version only generates functions for all endpoints, and i'm also using the non-union types for enums everywhere (which i guess was removed in an older version of your fork).

Also, i have this very specific setup where i need to have two input files to generate two separate output SDKs, which i couldn't get to work in the newer version... Only one of the output targets were being written, and it looked like both inputs were being "merged" in this single output for some reason. I didn't look much into it because the big refactor was already a deal breaker for me at the time.

But i honestly think you guys are doing awesome work, i just couldn't be bothered to migrate since i have very little issues with the current API and generated clients.

**Comment by mrlubos** [comment]
@Lamarcke is your project public? I could have a look... the "old style" API will make a comeback eventually. It might not be a perfect 1:1, but it should be way easier to migrate than rewriting everything. The rest of the issues I am vaguely familiar with, but would need to inspect further

**Comment by Lamarcke** [comment]
> @Lamarcke is your project public? I could have a look... the "old style" API will make a comeback eventually. It might not be a perfect 1:1, but it should be way easier to migrate than rewriting everything. The rest of the issues I am vaguely familiar with, but would need to inspect further

Basically these 3:
https://github.com/game-node-app/game-node-web (this one is the biggest)
https://github.com/game-node-app/game-node-admin
https://github.com/game-node-app/game-node-mobile

All of the three follow the same style, and the clients are generated in this file (which is replicated in all of them):
https://github.com/game-node-app/game-node-web/blob/main/src/wrapper/generate.js
